services/weather_handler.go

```diff
@@ -130,33 +136,52 @@ func (self *weatherHandler) HandleForecast(q string, days int32, cred string) (*
 			return nil, err
 		}
 
+		data := self.formForecastData(&day)
+		forecastOutput.Forecastday = append(forecastOutput.Forecastday, *data)
 	}
 
-	return forecast, nil
+	return &forecastOutput, nil
 }
 
 func (self *weatherHandler) formCurrentData(current *api.Current) *api.Current {
 
-	var outputData *api.Current
-
-	outputData.TempC = current.TempC
-	outputData.Condition = current.Condition
-	outputData.LastUpdated = current.LastUpdated
-	outputData.Cloud = current.Cloud
-	outputData.IsDay = current.IsDay
+	var outputData = &api.Current{
+		LastUpdated: current.LastUpdated,
+		TempC:       current.TempC,
+		Condition:   current.Condition,
+		Cloud:       current.Cloud,
+		IsDay:       current.IsDay,
+	}
 
 	return outputData
 }
 
-// func (self *weatherHandler) formForecastData(current *api.Forecast) *api.Current {
-//
-//	var outputData *api.Fo
-//
-//	outputData.TempC = current.TempC
-//	outputData.Condition = current.Condition
-//	outputData.LastUpdated = current.LastUpdated
-//	outputData.Cloud = current.Cloud
-//	outputData.IsDay = current.IsDay
-//
-//	return outputData
-// }
+func (self *weatherHandler) formForecastData(data *api.ForecastForecastday) *api.ForecastForecastday {
+
+	day := &api.ForecastDay{
+		Condition:         data.Day.Condition,
+		AvgtempC:          data.Day.AvgtempC,
+		MaxtempC:          data.Day.MaxtempC,
+		MintempC:          data.Day.MintempC,
+		DailyChanceOfRain: data.Day.DailyChanceOfRain,
+	}
+
+	hour := &[]api.ForecastHour{}
```

Comment:
Is there a particular reason why this is a pointer to a slice? I believe a normal slice will work fine here. You can have it be a slice of pointers:
```go
var hour []*api.ForecastHour
```

and then just append like this:
```go
hour = append(hour, x)
```